StudentApplication.java

```diff
@@ -0,0 +1,191 @@
+    }
+    public abstract void execute(Scanner scanner);
+    public abstract String getLabel();
+}
+
+class AddStudentMenuItem extends MenuItem {
+    public AddStudentMenuItem(StudentService studentService) { super(studentService); }
+    @Override
+    public void execute(Scanner scanner) {
+        System.out.print("Enter ID: ");
+        String id = scanner.next();
+        System.out.print("Enter Name: ");
+        String name = scanner.next();
+        System.out.print("Enter Age: ");
+        int age = scanner.nextInt();
+        studentService.addStudent(id, name, age);
+        System.out.println("Student added successfully.");
+    }
+    @Override
+    public String getLabel() { return "Add Student"; }
+}
+
+class UpdateStudentMenuItem extends MenuItem {
+    public UpdateStudentMenuItem(StudentService studentService) { super(studentService); }
+    @Override
+    public void execute(Scanner scanner) {
+        System.out.print("Enter Student ID to Update: ");
+        String id = scanner.next();
+        System.out.print("Enter New Name: ");
+        String name = scanner.next();
+        System.out.print("Enter New Age: ");
+        int age = scanner.nextInt();
+        studentService.updateStudent(id, name, age);
+        System.out.println("Student updated successfully.");
+    }
+    @Override
+    public String getLabel() { return "Update Student"; }
+}
+
+class RemoveStudentMenuItem extends MenuItem {
+    public RemoveStudentMenuItem(StudentService studentService) { super(studentService); }
+    @Override
+    public void execute(Scanner scanner) {
+        System.out.print("Enter Student ID to Remove: ");
+        String id = scanner.next();
+        studentService.removeStudent(id);
+        System.out.println("Student removed successfully.");
+    }
+    @Override
+    public String getLabel() { return "Remove Student"; }
+}
+
+class ShowStudentsMenuItem extends MenuItem {
+    public ShowStudentsMenuItem(StudentService studentService) { super(studentService); }
+    @Override
+    public void execute(Scanner scanner) {
+        List<Student> students = studentService.getAllStudents();
+        if (students.isEmpty()) {
+            System.out.println("No students available.");
+        } else {
+            students.forEach(System.out::println);
+        }
+    }
+    @Override
+    public String getLabel() { return "Show Students"; }
+}
+
+// Main 
+public class StudentApplication {
+    public static void main(String[] args) {
+        StudentDao studentDao = new StudentDaoImpl();
+        StudentService studentService = new StudentService(studentDao);
+        List<MenuItem> menuItems = List.of(
+                new AddStudentMenuItem(studentService),
+                new UpdateStudentMenuItem(studentService),
+                new RemoveStudentMenuItem(studentService),
+                new ShowStudentsMenuItem(studentService)
+        );
+        Scanner scanner = new Scanner(System.in);
+        while (true) {
```

Comment:
Maybe there is a better condition instead of `while (true) { ... }`. Try to find a different condition - check the menu, there is also an Exit menu item.